stdlib/reflect/src/main/java/org/ballerinalang/stdlib/reflect/nativeimpl/AbstractAnnotationReader.java

```diff
@@ -55,10 +55,14 @@ private BRefValueArray createAnnotationStructArray(Context context, BValue map)
         BMap<String, BValue> annotationMap = (BMap<String, BValue>) map;
         long index = 0;
         for (String key : annotationMap.keySet()) {
-            final String annotaionQName = key.split("\\$")[0];
-            final String[] qNameParts = annotaionQName.split(":");
+            final String annotationQName = key.split("\\$")[0];
+            final String annotationName = annotationQName.substring(annotationQName.lastIndexOf(":") + 1);
+            final String pkgQName = annotationQName.substring(0, annotationQName.lastIndexOf(":"));
+            final String[] pkgQNameParts = pkgQName.split(":");
```

Comment:
Shall we use `Names.VERSION_SEPARATOR.value`? would be easy to maintain in the long run.